### PR TITLE
Implement validate invocation

### PR DESCRIPTION
**Pull request by jakopako**

### CODE REVIEW

**Comment by zemberdotnet** [comment]
Hey @jakopako :wave: Thank you for your contribution! :tada: I'm going to take a look at this now.

File: src/config.ts [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+  const client = createFalconAPIClient(instance.config, logger);
+  try {
+    await client.authenticate();
+  } catch (err) {
+    throw new IntegrationProviderAuthenticationError({
+      cause: err,
+      endpoint: 'https://api.crowdstrike.com/oauth2/token',
+      status: 'failed',
+      statusText: 'auth error',
```
Comment (zemberdotnet) [inline src/config.ts:60]: We usually refer directly to the `status` and `statusText` properties on the `err` object. In this case the error object is being thrown here: https://github.com/JupiterOne/graph-crowdstrike/blob/50de1c1dac6a3a4594722082f903d4992cb3b23c/src/crowdstrike/FalconAPIClient.ts#L273-L278
The error will have `code` property which is the same as the `status` property. The error is missing the `statusText` property, so I think we should add it in.

The `ApiResponse` object it is using to assign properties to the error is created here:
https://github.com/JupiterOne/graph-crowdstrike/blob/50de1c1dac6a3a4594722082f903d4992cb3b23c/src/crowdstrike/FalconAPIClient.ts#L301-L306

I think that we should make this change:
**`src/crowdstrike/FalconAPIClient.ts`**
```diff
    if (apiResponse.status >= 400) {
      const err = new Error(
        `API request error for ${info}: ${apiResponse.statusText}`,
      );
-     Object.assign(err, { code: apiResponse.status });
+     Object.assign(err, {
+        code: apiResponse.status,
+        statusText: apiResponse.statusText,
+      });
      throw err;
    }

    return apiResponse.response.json();
```

Then we could make this change here:
```suggestion
      status: err.code,
      statusText: err.statusText,
```

**Review by zemberdotnet — CHANGES_REQUESTED** [review]
Awesome work here! These are great changes. There is one change that need to be made to the thrown error object in `validateInvocation`. I tried to explain the needed changes, but let me know if you have any questions.

File: src/config.ts (hunk 1 above)
Comment (jakopako) [reply src/config.ts]: Thanks for the review! I'll update the code accordingly.

**Comment by jakopako** [comment]
> Hi @jakopako! Thanks for your contributions! I've approved and merged this to main. 🎉

Nice! Thanks for accepting my changes!

**Pull request merged**